src/Nethermind/Nethermind.Optimism/OptimismPostMergeBlockProducer.cs

```diff
@@ -52,45 +51,27 @@ public OptimismPostMergeBlockProducer(
         _specHelper = specHelper;
     }
 
-    protected override Block CreateEmptyBlock(BlockHeader parent, PayloadAttributes? payloadAttributes = null)
+    protected override BlockToProduce PrepareBlock(BlockHeader parent, PayloadAttributes? payloadAttributes = null, IBlockProducer.Flags flags = IBlockProducer.Flags.None)
     {
         OptimismPayloadAttributes attrs = (payloadAttributes as OptimismPayloadAttributes)
             ?? throw new InvalidOperationException("Payload attributes are not set");
 
-        BlockHeader blockHeader = base.PrepareBlockHeader(parent, attrs);
-
-        IEnumerable<Transaction> txs = _payloadAttrsTxSource.GetTransactions(parent, attrs.GasLimit, attrs);
-
-        Block block = new(blockHeader, txs, Array.Empty<BlockHeader>(), payloadAttributes?.Withdrawals);
-
-        if (_producingBlockLock.Wait(BlockProductionTimeoutMs))
+        BlockToProduce blockToProduce = base.PrepareBlock(parent, payloadAttributes, flags);
+        if ((flags & IBlockProducer.Flags.EmptyBlock) != 0)
         {
-            try
-            {
-                if (TrySetState(parent.StateRoot))
-                {
-                    return ProcessPreparedBlock(block, null) ?? throw new EmptyBlockProductionException("Block processing failed");
```

Comment:
Optimism empty blocks are processed, do we process empty blocks now? (we might after withdrawals and other system contracts)

Reply:
Yes, before it was like that for ethereum too.